cgen error for an if expression, with a panic() call in the else branch, assigned to a `?string` variable

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.8 7565841.72b4804
OS: linux, Ubuntu 20.04.6 LTS
Processor: 2 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /space/v/oo/v
vexe mtime: 2024-10-10 01:31:53

vroot: OK, value: /space/v/oo
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.46.0
Git vroot status: 0.4.8-75-g72b48047
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg x.v`
```v
mut x := ?string(none)
x = if true {
	''
} else {
	panic('')
}
println(x)

```

**What did you expect to see?**

a compiled program

**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/x.01J9T0VW2FV63VCS5NVGQ8NEYN.tmp.c:13450: error: declaration of an array of incomplete type elements
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -1926,9 +1926,14 @@ fn (mut g Gen) stmts_with_tmp_var(stmts []ast.Stmt, tmp_var string) bool {
 								g.fn_decl.return_type.clear_flag(.option)
 							}
 							styp = g.base_type(ret_typ)
-							g.write('_option_ok(&(${styp}[]) { ')
-							g.expr_with_cast(stmt.expr, stmt.typ, ret_typ)
-							g.writeln(' }, (${option_name}*)(&${tmp_var}), sizeof(${styp}));')
+							if stmt.expr is ast.CallExpr && stmt.expr.name == 'panic' {
+								g.expr(stmt.expr)
+								g.writeln(';')
+							} else {
+								g.write('_option_ok(&(${styp}[]) { ')
+								g.expr_with_cast(stmt.expr, stmt.typ, ret_typ)
+								g.writeln(' }, (${option_name}*)(&${tmp_var}), sizeof(${styp}));')
+							}
 						}
 					}
 				}
@@ -1958,9 +1963,14 @@ fn (mut g Gen) stmts_with_tmp_var(stmts []ast.Stmt, tmp_var string) bool {
 						} else {
 							ret_typ := g.fn_decl.return_type.clear_flag(.result)
 							styp = g.base_type(ret_typ)
-							g.write('_result_ok(&(${styp}[]) { ')
-							g.expr_with_cast(stmt.expr, stmt.typ, ret_typ)
-							g.writeln(' }, (${result_name}*)(&${tmp_var}), sizeof(${styp}));')
+							if stmt.expr is ast.CallExpr && stmt.expr.name == 'panic' {
+								g.expr(stmt.expr)
+								g.writeln(';')
+							} else {
+								g.write('_result_ok(&(${styp}[]) { ')
+								g.expr_with_cast(stmt.expr, stmt.typ, ret_typ)
+								g.writeln(' }, (${result_name}*)(&${tmp_var}), sizeof(${styp}));')
+							}
 						}
 					}
 				}

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -1926,7 +1926,7 @@ fn (mut g Gen) stmts_with_tmp_var(stmts []ast.Stmt, tmp_var string) bool {
 								g.fn_decl.return_type.clear_flag(.option)
 							}
 							styp = g.base_type(ret_typ)
-							if stmt.expr is ast.CallExpr && stmt.expr.name == 'panic' {
+							if stmt.expr is ast.CallExpr && stmt.expr.is_noreturn {
 								g.expr(stmt.expr)
 								g.writeln(';')
 							} else {
@@ -1963,7 +1963,7 @@ fn (mut g Gen) stmts_with_tmp_var(stmts []ast.Stmt, tmp_var string) bool {
 						} else {
 							ret_typ := g.fn_decl.return_type.clear_flag(.result)
 							styp = g.base_type(ret_typ)
-							if stmt.expr is ast.CallExpr && stmt.expr.name == 'panic' {
+							if stmt.expr is ast.CallExpr && stmt.expr.is_noreturn {
 								g.expr(stmt.expr)
 								g.writeln(';')
 							} else {

--- a/vlib/v/tests/options/option_if_expr_with_panic_call_test.v
+++ b/vlib/v/tests/options/option_if_expr_with_panic_call_test.v
@@ -1,0 +1,10 @@
+fn test_option_if_expr_with_panic_call() {
+	mut x := ?string(none)
+	x = if true {
+		''
+	} else {
+		panic('')
+	}
+	println(x)
+	assert true
+}

